### PR TITLE
Ministers index page - SITES-374

### PR DESCRIPTION
**Pull request by micapam**
Basic ministers index page. 
There are some hacky bits in the decorator. Apparently (from discussions with agency team) this page may change so I didn't want to go too deep into the logic until the requirements are stable...

image

### CODE REVIEW

**Comment by maxious** [comment]
LGTM

File: app/decorators/minister_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      name_parts.first.strip
+    end
+  end
+
+  def ministry_or_title
+    name_parts.last
+  end
+
+  #TODO clean this up once requirements are clear
+  def sort_order
```
Comment (coderdan) [inline app/decorators/minister_decorator.rb]: Whoah. Let me bow down to you oh master of the hacks! hahaha :p

File: spec/controllers/ministers_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+require 'rails_helper'
+
+RSpec.describe MinistersController, type: :controller do
```
Comment (coderdan) [inline spec/controllers/ministers_controller_spec.rb]: render_views ?

File: spec/controllers/ministers_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  describe 'GET #index' do
+    let!(:health) { Fabricate(:minister, name: 'Minister for Health') }
+    let!(:pm) { Fabricate(:minister, name: 'Prime Minister') }
+    let!(:comms) { Fabricate(:minister, name: 'Minister for Communications') }
+
+    before { get :index }
+
+    it { expect(subject.ministers).to eq [pm, comms, health] }
```
Comment (coderdan) [inline spec/controllers/ministers_controller_spec.rb]: You should probably test that the controller render was successful.

    it { is_expected.to be_success }

File: config/breadcrumbs.rb [hunk 4]
```diff
@@ -7,6 +7,11 @@
   parent :public_node, Node.root
 end
 
+crumb :ministers do
+  link 'Ministers', ministers_path
+  parent :public_node, Node.root
+end
+
```
Comment (tessereth) [inline config/breadcrumbs.rb]: Should the ministers pages have `:ministers` as the parent breadcrumb?

**Comment by coderdan** [comment]
A couple of little things that we should discuss.

File: spec/controllers/ministers_controller_spec.rb (hunk 2 above)
Comment (tessereth) [inline spec/controllers/ministers_controller_spec.rb]: Don't need it for the tests as written.

**Comment by tessereth** [comment]
Just a minor question about breadcrumbs but LGTM

**Comment by nicholasgkdto** [comment]
The bread crumb will be `Home > Minister for Communications`  not `home > Ministers > Minister for Communications`

**Comment by micapam** [comment]
@nicholasgkdto - I'm not clear as to whether you're saying that's a problem, or that's how the breadcrumbs _should_ appear?

File: spec/controllers/ministers_controller_spec.rb (hunk 2 above)
Comment (micapam) [inline spec/controllers/ministers_controller_spec.rb]: We need to discuss this as a team methinks....

File: app/decorators/minister_decorator.rb (hunk 1 above)
Comment (micapam) [inline app/decorators/minister_decorator.rb]: LOL. This is quite awful I know!

**Comment by nicholasgkdto** [comment]
The page shown above is correct.  the list of minsters will be `Home > Ministers`.  so no problem there.  it is when you open a minsters page, it will be like  `Home > Minister for Monkeys`.  Does that make sense.

**Comment by micapam** [comment]
@nicholasgkdto, I'm afraid it's still not clear (to me) which of these two things you're saying: 
1. Breadcrumbs for minister page should be `Home > Ministers > Minister for Monkeys`
2. Breadcrumbs for minister page should be `Home > Minister for Monkeys`

**Comment by nicholasgkdto** [comment]
second one is correct,  see the links below.
[List page](http://govau-agency.apps.staging.digital.gov.au/ministers.html)  
[minster page](http://govau-agency.apps.staging.digital.gov.au/comms-minister/index.html)

**Comment by tessereth** [comment]
Thanks @nicholasgkdto. That means the original breadcrumbs config is correct and my comment can be ignored.

**Comment by micapam** [comment]
So I have the requisite lgtms but waiting on @coderdan as he said there is stuff to discuss.

**Comment by ltankey** [comment]
Please update the changelog.

**Comment by micapam** [comment]
@ltankey Is that OK now? (My first time updating the changelog...)

**Comment by ltankey** [comment]
Thanks @micapam, that's good. We're still settling in to a language style for changelog updates, which I think will drift more towards 'A list of ministers on an index page' but I think your message is fine! Thanks!

**Comment by ltankey** [comment]
LGTM

**Comment by micapam** [comment]
bump

**Pull request merged**